pyro/infer/mcmc/api.py

```diff
@@ -333,6 +334,7 @@ def __init__(self, kernel, num_samples, warmup_steps=None, initial_params=None,
             self.sampler = _UnarySampler(kernel, num_samples, self.warmup_steps, disable_progbar,
                                          initial_params=initial_params, hook=hook_fn)
 
+    @poutine.block
```

Comment:
@neerajprad why we need `block` here?

Reply:
This is so that if we are tracing a model using the generated samples from MCMC, we can run inference within a `poutine.trace` contextmanager without capturing the traces from the inference method itself. See `test_abstract_infer.test_nesting`. That test will fail without this. This was also the behavior in `TracePosterior` class which is now deprecated.

Reply:
Understood, thanks for explaining!